contracts/controllers/DelegatedController.sol

```diff
@@ -0,0 +1,187 @@
+        return Strategy(strategies[_vault]).balanceOf();
+    }
+
+    function withdrawAll(address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        // WithdrawAll sends 'want' to 'vault'
+        Strategy(_strategy).withdrawAll();
+    }
+
+    function inCaseTokensGetStuck(address _token, uint256 _amount) external {
+        require(msg.sender == governance, "!governance");
+        IERC20(_token).safeTransfer(governance, _amount);
+    }
+
+    function inCaseStrategyGetStruck(address _strategy, address _token) external {
+        require(msg.sender == governance, "!governance");
+        Strategy(_strategy).withdraw(_token);
+        IERC20(_token).safeTransfer(governance, IERC20(_token).balanceOf(address(this)));
+    }
+
+    function getExpectedReturn(
+        address _strategy,
+        address _token,
+        uint256 parts
+    ) external view returns (uint256 expected) {
+        uint256 _balance = IERC20(_token).balanceOf(_strategy);
+        address _want = Strategy(_strategy).want();
+        (expected, ) = OneSplitAudit(onesplit).getExpectedReturn(_token, _want, _balance, parts, 0);
+    }
+
+    function claimInsurance(address _vault) external {
+        require(msg.sender == governance, "!governance");
+        DelegatedVault(_vault).claimInsurance();
+    }
+
+    // Only allows to withdraw non-core strategy tokens ~ this is over and above normal yield
+    function delegatedHarvest(address _strategy, uint256 parts) external {
+        // This contract should never have value in it, but just incase since this is a public call
+        address _have = Strategy(_strategy).want();
+        uint256 _before = IERC20(_have).balanceOf(address(this));
+        Strategy(_strategy).skim();
+        uint256 _after = IERC20(_have).balanceOf(address(this));
+        if (_after > _before) {
+            uint256 _amount = _after.sub(_before);
+            address _want = DelegatedVault(vaults[_strategy]).token();
+            uint256[] memory _distribution;
+            uint256 _expected;
+            _before = IERC20(_want).balanceOf(address(this));
+            IERC20(_have).safeApprove(onesplit, 0);
+            IERC20(_have).safeApprove(onesplit, _amount);
+            (_expected, _distribution) = OneSplitAudit(onesplit).getExpectedReturn(_have, _want, _amount, parts, 0);
+            OneSplitAudit(onesplit).swap(_have, _want, _amount, _expected, _distribution, 0);
+            _after = IERC20(_want).balanceOf(address(this));
+            if (_after > _before) {
+                _amount = _after.sub(_before);
+                uint256 _reward = _amount.mul(split).div(max);
+                IERC20(_want).safeTransfer(vaults[_strategy], _amount.sub(_reward));
+                IERC20(_want).safeTransfer(rewards, _reward);
+            }
+        }
+    }
+
+    // Only allows to withdraw non-core strategy tokens ~ this is over and above normal yield
+    function harvest(
```

Comment:
Need to understand the difference between this and `delegatedHarvest`, and whether there's an oppurtunity for de-duplication